File: Solutions/225- Implement Stack using Queues/Solution.py
class MyStack:

    def __init__(self):
        self.main_queue = []
        self.temp_queue = []

    def push(self, x: int) -> None:
        self.main_queue.append(x)

    def pop(self) -> int:
        return self.main_queue.pop()

    def top(self) -> int:
        return self.main_queue[-1]

    def empty(self) -> bool:
        return len(self.main_queue) == 0

File: Solutions/225- Implement Stack using Queues/Solution.java
import java.util.LinkedList;
import java.util.Queue;

class MyStack {
    Queue<Integer> main_queue;
    Queue<Integer> temp_queue;
    public MyStack() {
        main_queue = new LinkedList<>();
        temp_queue = new LinkedList<>();
    }

    public void push(int x) {
        main_queue.add(x);
    }

    public int pop() {
        var size = main_queue.size();

        for (int i = 0; i < size - 1; i++) {
            temp_queue.add(main_queue.remove());
        }
        var number = main_queue.remove();

        size = temp_queue.size();

        for (int i = 0; i < size; i++) {
            main_queue.add(temp_queue.remove());
        }

        return number;
    }

    public int top() {
        var size = main_queue.size();
        for (int i = 0; i < size - 1; i++) {
            temp_queue.add(main_queue.remove());
        }

        var number = main_queue.remove();

        temp_queue.add(number);

        size = temp_queue.size();

        for (int i = 0; i < size; i++) {
            main_queue.add(temp_queue.remove());
        }

        return number;
    }

    public boolean empty() {
        return main_queue.isEmpty();
    }
}